File: src/main/java/net/atos/zac/app/zaken/converter/RESTBesluittypeConverter.java
/*
 * SPDX-FileCopyrightText: 2022 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.zac.app.zaken.converter;

import java.net.URI;
import java.util.List;

import jakarta.inject.Inject;

import net.atos.client.zgw.ztc.ZTCClientService;
import net.atos.client.zgw.ztc.model.generated.BesluitType;
import net.atos.zac.app.zaken.model.RESTBesluittype;
import net.atos.zac.util.UriUtil;

public class RESTBesluittypeConverter {

    @Inject
    private ZTCClientService ztcClientService;

    public RESTBesluittype convertToRESTBesluittype(final BesluitType besluittype) {
        final RESTBesluittype restBesluittype = new RESTBesluittype();
        restBesluittype.id = UriUtil.uuidFromURI(besluittype.getUrl());
        restBesluittype.naam = besluittype.getOmschrijving();
        restBesluittype.toelichting = besluittype.getToelichting();
        restBesluittype.informatieobjecttypen = besluittype.getInformatieobjecttypen().stream().toList();
        return restBesluittype;
    }

    public RESTBesluittype convertToRESTBesluittype(final URI besluittypeURI) {
        return convertToRESTBesluittype(ztcClientService.readBesluittype(besluittypeURI));
    }

    public List<RESTBesluittype> convertToRESTBesluittypes(final List<BesluitType> besluittypes) {
        return besluittypes.stream()
                .map(this::convertToRESTBesluittype)
                .toList();
    }

}

File: src/main/kotlin/net/atos/zac/app/zaken/converter/RESTBesluittypeConverter.kt
/*
 * SPDX-FileCopyrightText: 2022 Atos, 2024 Lifely
 * SPDX-License-Identifier: EUPL-1.2+
 */
package net.atos.zac.app.zaken.converter

import jakarta.inject.Inject
import net.atos.client.zgw.ztc.ZTCClientService
import net.atos.client.zgw.ztc.model.generated.BesluitType
import net.atos.zac.app.zaken.model.RESTBesluittype
import net.atos.zac.util.UriUtil
import java.net.URI

class RESTBesluittypeConverter {
    @Inject
    private lateinit var ztcClientService: ZTCClientService

    fun convertToRESTBesluittype(besluittypeURI: URI): RESTBesluittype {
        return convertToRESTBesluittype(ztcClientService.readBesluittype(besluittypeURI))
    }

    fun convertToRESTBesluittypes(besluittypes: List<BesluitType>): List<RESTBesluittype> {
        return besluittypes.stream()
            .map { besluittype -> this.convertToRESTBesluittype(besluittype) }
            .toList()
    }

    private fun convertToRESTBesluittype(besluittype: BesluitType) = RESTBesluittype(
        id = UriUtil.uuidFromURI(besluittype.url),
        naam = besluittype.omschrijving,
        toelichting = besluittype.toelichting,
        informatieobjecttypen = besluittype.informatieobjecttypen.stream().toList()
    )
}
